File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        if n <= 0 or n > 20:
            return

        matrix = [[0 for _ in range(n)] for _ in range(n)]
        top,left = 0,0
        bottom,right = n-1,n-1
        count = 0

        while count != n*n:
            for col in range(left,right+1):
                matrix[top][col] = count+1
                count+=1
            top += 1

            for row in range(top,bottom+1):
                matrix[row][right] = count+1
                count+=1
            right -= 1

            if top <= bottom:
                for col in range(right,left-1,-1):
                    matrix[bottom][col]=count+1
                    count+=1
                bottom -= 1

            if left <= right:
                for row in range(bottom,top-1,-1):
                    matrix[row][left] = count+1
                    count+=1
                left += 1
        return matrix 

File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.java
class Solution {
    public int[][] generateMatrix(int n) {
        if(n <= 0 || n > 20){
            return null;
        }
        int[][] matrix = new int[n][n];
        int top = 0,left = 0;
        int bottom = n-1,right = n-1;
        int count = 0;

        while(count != (n*n)){
            for(int col=left;col < right + 1;col++){
                matrix[top][col] = count+1;
                count++;
            }
            top++;

            for(int row=top;row < bottom + 1;row++){
                matrix[row][right] = count+1;
                count++;
            }
            right--; 

            if(top <= bottom){
                for(int col=right;col > left - 1;col--){
                matrix[bottom][col] = count+1;
                count++;
            }
            bottom--;
            }

            if(left <= right){
                for(int row=bottom;row > top - 1;row--){
                matrix[row][left] = count+1;
                count++;
            }
            left++;
            }       
        }
        return matrix;
    }
}